build_tools/github_actions/lint_whitespace_checks.sh

```diff
@@ -34,12 +35,13 @@ if [[ $# -ne 0 ]] ; then
 fi
 
 get_source_files() {
-  git diff $BASE_BRANCH HEAD --name-only --diff-filter=d | grep '.*\.cpp$\|.*\.h$\|.*\.md$\|.*\.mlir$\|.*\.sh$\|.*\.td$\|.*\.txt$\|.*\.yml$\|.*\.yaml$' | xargs
+  git diff "$BASE_BRANCH" HEAD --name-only --diff-filter=d | grep '.*\.cpp$\|.*\.h$\|.*\.md$\|.*\.mlir$\|.*\.sh$\|.*\.td$\|.*\.txt$\|.*\.yml$\|.*\.yaml$' | xargs
 }
 echo "Checking whitespace:"
 echo "  $(get_source_files)"
 
 files_without_eof_newline() {
+  # shellcheck disable=SC2016
```

Comment:
another option:
> ShellCheck also does not warn about escaped expansions in double quotes:

Reply:
Ah yes, but that will do the wrong thing unfortunately, because Bash will try to expand a variable called `a`, which does not exist, so the `sed` command will be incorrect.